scripts/plotting/time_series.py

```diff
@@ -0,0 +1,630 @@
+                ax = _format_yaxis(ax, case_num, units[var], **vres)
+
+                #Set up legend
+                fig = _make_fig_legend(case_num, fig)
+
+                #Save plot
+                plot_name = plot_loc / f"{var}_{season}_TimeSeries_Mean.{plot_type}"
+                plt.savefig(plot_name, facecolor='w')
+                plt.close()
+
+                #Add plot to website (if enabled):
+                adfobj.add_website_data(plot_name, var, case, season=season,
+                                        plot_type="TimeSeries",
+                                        multi_case=multi_case)
+                #End for (cases)
+            #End for (season)
+        #End if (not in del_s)
+    #End for (var)
+    #End seasonally weighted
+
+    #Notify user that script has ended:
+    print("  ...time series have been generated successfully.")
+
+#Helper functions:
+##################
+
+def _load_dataset(fils):
+    if len(fils) == 0:
+        print("Input file list is empty.")
+        return None
+    elif len(fils) > 1:
+        return xr.open_mfdataset(fils, combine='by_coords')
+    else:
+        sfil = str(fils[0])
+        return xr.open_dataset(sfil)
+    #End if
+#End def
+
+########
+
+def _data_calcs(var, ts_ds, subset=None):
+    """
+    Function to calculate global weighted data
+    ----
+        - weight global data by latitude
+        - option for lat/lon subset
+        - returns
+            * month lengths
+            * climo years
+            * units
+    """
+
+    time = ts_ds['time']
+    time = xr.DataArray(ts_ds['time_bnds'].load().mean(dim='nbnd').values, dims=time.dims, attrs=time.attrs)
+    ts_ds['time'] = time
+    ts_ds.assign_coords(time=time)
+    ts_ds = xr.decode_cf(ts_ds)
+
+    if subset != None:
+        ts_ds = ts_ds.sel(lat=slice(subset["s"],subset["n"]), lon=slice(subset["w"],subset["e"])) 
+
+    data = ts_ds[var].squeeze()
+    month_length = data.time.dt.days_in_month
+    unit = data.units
+
+    #global weighting
+    w = np.cos(np.radians(data.lat))
+    avg = data.weighted(w).mean(dim=("lat","lon"))
+
+    yrs = np.unique([str(val.item().timetuple().tm_year).zfill(4) for _,val in enumerate(ts_ds["time"])])
+
+    return avg,month_length,yrs,unit
+
+########
+
+def seasonal_data(data, month_length):
+    """
+    Function to grab seasonal weighted data, grouped by season
+    -----
+        - DJF, MAM, JJA, and SON
+    """
+
+    weighted_mean = ((data * month_length).resample(time='QS-DEC').sum() /
+                          month_length.resample(time='QS-DEC').sum())
+
+    mdata_seasonal_mean = weighted_mean.groupby('time.season')
+
+    return mdata_seasonal_mean
+
+########
+
+def _get_seasonal_data(ts_var_list, all_case_names, case_ts_locs):
+    """
+    Gather seasonally weighted data
+    -----
+        - 
+    """
+
+    #Keep track of variables that have vertical levels for now
+    #There is probably a better way to ignore these vars - JR
+    del_s = []
```

Comment:
It would probably be slightly more performant if `del_s` was a set instead of a list, since the only thing it is being used for is to check if a variable is a member:

```suggestion
    del_s = set()
```